### DISCUSSION TITLE
I want to change the reference clock

### DISCUSSION THREAD

**Discussion by Usutatsu** [polarfire-soc/polarfire-soc-documentation#247] [How to] [answered]
I am currently developing with the Polarfire Soc Icicle kit using the latest version of the reference design.
The latest reference design uses a 50MHZ on-chip oscillator as input to a PLL in an FPGA fabric.
I would like to replace this 50MHZ oscillator with a 125MHz REFCKL input to the MSS.
Is this possible? And if possible, I would like to know how to implement it.

**Comment by sanjeevb-bansal** [comment, marked as answer]
Yes, this is possible. Pls use the NWPLL for the ref clock of the MSS then it is possible. This can be configured in PFSoC MSS Configurator v2022.2 . Pls go to "clocks" tab use the drop down "MSS PLL Ref Clock Source" and select "NW PLL ..". 

Reference:-
https://ww1.microchip.com/downloads/aemDocuments/documents/FPGA/ProductDocuments/ReferenceManuals/PolarFire_SoC_FPGA_MSS_Technical_Reference_Manual_VC.pdf
https://coredocs.s3.amazonaws.com/Libero/2021_1/pfsoc_mss_configurator_ug.pdf ( see section 2.3.1)
https://ww1.microchip.com/downloads/aemDocuments/documents/FPGA/ProductDocuments/UserGuides/Microchip_PolarFire_FPGA_and_PolarFire_SoC_FPGA_Clocking_Resources_User_Guide_VB.pdf

_"The PolarFire SoC MSS has a dedicated clock controller for generating clocks to all the MSS sub-blocks for correct operation and ynchronous communication with the user logic in the FPGA fabric. The MSS clock controller includes dedicated PLLs (MPLLs, DDR PLL, and SGMII PLL) for MSS clocking. **The base clock for these PLLs comes either from a dedicated I/O (REFCLK) from Bank5 or one of the NW PLL outputs—OUT2 or OUT3. This dedicated I/O can be connected to a clock source, which can supply 100 MHz or 125 MHz clock.**"_

**Reply by Usutatsu** [reply]
Thank you for your comment.

I understood that I can use the output of the NW PLL as an input to the MSS Clock controller.

What I really want to do this time is to use the REFCLK from Bank5 as an input to the PLL (CCC) of the FPGA fabric.
Does this mean that this is not possible?

**Reply by sanjeevb-bansal** [reply]
I double checked, yes it is possible  to use the REFCLK from Bank5 as an input to the PLL (CCC) of the FPGA fabric. MSS Configurator tool supports this configuration as well.

**Reply by Usutatsu** [reply]
Thank you for confirming.
I underdtood that can use REFCLK from Bank5 as a source clock of FPGA fabric PLL(CCC).

You taught me MSS configuration tool supports this configuration.
However, I could not find that configuration for inputting REFCLK to PLL(CCC).
Can you tell me how to set this up?

**Reply by Usutatsu** [reply]
Incidentally, is it possible to use more than one PLL in the CCC at the same time?
For example, using CCC_NW and CCC_SW at the same time.
Is it possible to input the clock generated by CCC_NW into MSS_PLL and output the clock generated by CCC_SW out of the fabric through GPIOs?

**Reply by sanjeevb-bansal** [reply]
I hope you able to launch PFSoC MSS Configurator. This is automatically installed along with Libero installation. Pls see the image below:-

![image](https://user-images.githubusercontent.com/77011968/203743250-3b1df254-7d9d-4239-82b6-f065520d13ac.png)

For 2nd part I see that you have created a new discussion already. https://github.com/orgs/polarfire-soc/discussions/251 

Let's carry this discussion to the new thread you have created.

**Comment by sanjeevb-bansal** [comment]
On the side it is good to though below document which describes our "PolarFire SoC Software Tool Flow" :-
https://github.com/polarfire-soc/polarfire-soc-documentation/blob/master/software-development/polarfire-soc-software-tool-flow.md
